app/core/log.py

```diff
@@ -1,13 +1,46 @@
 import logging
 import logging.config
 import queue
+from enum import Enum
 from logging.handlers import QueueHandler, QueueListener
 from pathlib import Path
 from typing import Any
 
+import uvicorn
+
 from app.core.config import Settings
 
 
+class ColoredConsoleFormatter(uvicorn.logging.DefaultFormatter):
+    class ConsoleColors(str, Enum):
+        """Colors can be found here: https://talyian.github.io/ansicolors/"""
+
+        DEBUG = "\033[38;5;12m"
+        INFO = "\033[38;5;10m"
+        WARNING = "\033[38;5;11m"
+        ERROR = "\033[38;5;9m"
+        CRITICAL = "\033[38;5;1m"
+        BOLD = "\033[1m"
+        END = "\033[0m"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(datefmt="%d-%b-%y %H:%M:%S")
+
+    def format(self, record: logging.LogRecord) -> str:
+        fmt = (
+            "%(asctime)s - %(name)s - "
+            + self.ConsoleColors.BOLD
+            + "%(levelname)s"
+            + self.ConsoleColors.END
+            + " - "
+            + self.ConsoleColors[record.levelname].value
+            + "%(message)s"
+            + self.ConsoleColors.END
+        )
+        formatter = logging.Formatter(fmt, self.datefmt)
+        return formatter.format(record)
```

Comment:
I'm not sure about this idea, but could this be better to prevent a new formatter instantiation for each `format` call?
```suggestion
    class ConsoleColors(str, Enum):
        """Colors can be found here: https://talyian.github.io/ansicolors/"""

        DEBUG = "\033[38;5;12m"
        INFO = "\033[38;5;10m"
        WARNING = "\033[38;5;11m"
        ERROR = "\033[38;5;9m"
        CRITICAL = "\033[38;5;1m"
        BOLD = "\033[1m"
        END = "\033[0m"

    def __init__(self, *args, **kwargs):
        super().__init__(datefmt="%d-%b-%y %H:%M:%S")
        
        self.formatters = {}
        
        for level in [logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR, logging.CRITICAL]:
            fmt = (
                "%(asctime)s - %(name)s - "
                + self.ConsoleColors.BOLD
                + "%(levelname)s"
                + self.ConsoleColors.END
                + " - "
                + self.ConsoleColors[level]
                + "%(message)s"
                + self.ConsoleColors.END
            )
            level_formatter = logging.Formatter(fmt, self.datefmt)
            self.formatters[level] = level_formatter

    def format(self, record: logging.LogRecord) -> str:
        formatter = self.formatters.get(record.levelno, self.formatters[logging.ERROR])
        return formatter.format(record)
```